demos/coreMQTT/mqtt_demo_plaintext.c

```diff
@@ -0,0 +1,969 @@
+ * with an exponential backoff mechanism.
+ */
+int RunCoreMqttPlaintextDemo( bool awsIotMqttMode,
+                              const char * pIdentifier,
+                              void * pNetworkServerInfo,
+                              void * pNetworkCredentialInfo,
+                              const void * pNetworkInterface )
+{
+    uint32_t ulPublishCount = 0U, ulTopicCount = 0U;
+    const uint32_t ulMaxPublishCount = 5UL;
+    NetworkContext_t xNetworkContext = { 0 };
+    MQTTContext_t xMQTTContext;
+    MQTTStatus_t xMQTTStatus;
+    BaseType_t xDemoStatus;
+    TransportSocketStatus_t xNetworkStatus;
+    uint32_t ulDemoRunCount = 0;
+    BaseType_t xIsConnectionEstablished = pdFALSE;
+
+    /* Upon return, EXIT_SUCCESS will indicate a successful demo execution.
+     * EXIT_FAILURE will indicate some failures occurred during execution. The
+     * user of this demo must check the logs for any failure codes. */
+    int lReturnStatus = EXIT_SUCCESS;
+
+    /* Remove compiler warnings about unused parameters. */
+    ( void ) awsIotMqttMode;
+    ( void ) pIdentifier;
+    ( void ) pNetworkServerInfo;
+    ( void ) pNetworkCredentialInfo;
+    ( void ) pNetworkInterface;
+
+    ulGlobalEntryTimeMs = prvGetTimeMs();
+
+    for( ; ulDemoRunCount < democonfigMQTT_MAX_DEMO_COUNT; ulDemoRunCount++ )
+    {
+        /****************************** Connect. ******************************/
+
+        /* Attempt to connect to the MQTT broker. If connection fails, retry after
+         * a timeout. The timeout value will be exponentially increased until the
+         * maximum number of attempts are reached or the maximum timeout value is
+         * reached. The function returns a failure status if the TCP connection
+         * cannot be established to the broker after the configured number of
+         * attempts. */
+        xNetworkStatus = prvConnectToServerWithBackoffRetries( &xNetworkContext );
+
+        if( xNetworkStatus != TRANSPORT_SOCKET_STATUS_SUCCESS )
+        {
+            lReturnStatus = EXIT_FAILURE;
+        }
+        else
+        {
+            /* Set a flag indicating a TCP connection exists. This is done to
+             * disconnect if the loop exits before disconnection happens. */
+            xIsConnectionEstablished = pdTRUE;
+        }
+
+        if( lReturnStatus == EXIT_SUCCESS )
+        {
+            /* Sends an MQTT Connect packet over the already connected TCP socket,
+             * and waits for connection acknowledgment (CONNACK) packet. */
+            LogInfo( ( "Creating an MQTT connection to %s.\r\n", democonfigMQTT_BROKER_ENDPOINT ) );
+            xDemoStatus = prvCreateMQTTConnectionWithBroker( &xMQTTContext, &xNetworkContext );
+
+            if( xDemoStatus != pdPASS )
+            {
+                /* Any errors from direct MQTT library calls are logged in the
+                 * private functions. */
+                lReturnStatus = EXIT_FAILURE;
```

Comment:
Instead of having this logic of converting `pdPASS` to `EXIT_FAILURE` through this function, we can just determine the return code of this function at the end with a ternary condition as 
`return ( xDemoStatus  == pdPASS ) ? EXIT_SUCCESS : EXIT_FAILURE`

Reply:
I will update this to shorted the lines of code.

Reply:
I will use only pdPASS and pdFAIL to short some lines.